Content.Shared/Turrets/DeployableTurretComponent.cs

```diff
@@ -0,0 +1,172 @@
+using Content.Shared.Damage.Prototypes;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Turrets;
+
+/// <summary>
+/// Attached to turrets that can be toggled between an inactive and active state
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+[Access(typeof(SharedDeployableTurretSystem))]
+public sealed partial class DeployableTurretComponent : Component
+{
+    /// <summary>
+    /// Whether the turret is toggled 'on' or 'off'
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool Enabled = false;
+
+    /// <summary>
+    /// Whether the turret is currently broken
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool Broken = false;
+
+    /// <summary>
+    /// Whether the turret is currently powered (required until power is predicted)
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool Powered = false;
+
```

Comment:
We have components for this, components shouldn't be storing it.

Reply:
The compstate is networked. Even if the power change itself isn't predicted the client will still know if it's powered or not 99% of the time.